### DISCUSSION TITLE
linux/v2raya

### DISCUSSION THREAD

**Discussion by giscus** [LaoYutang/blogGiscus#14] [Announcements]
# linux/v2raya

在软路由或者nas上使用docker搭建一个订阅代理服务，快速启动，界面简单易用。

https://blog.laoyutang.cn/linux/v2raya.html

**Comment by shnehna** [comment]
难受 不行啊
3	关闭退出软件	
127.0.0.1:443

SS(aes-256-gcm)	

2	或此软件不支持	
127.0.0.1:443

SS(aes-256-gcm)	

4	查看使用教程	
127.0.0.1:443

SS(aes-256-gcm)	

1	此软件需要更新	
127.0.0.1:443

SS(aes-256-gcm)	

5	第一个文档解决	
127.0.0.1:443

SS(aes-256-gcm)

**Reply by LaoYutang** [reply]
你在说啥啊，为啥每句话我都看得懂，连起来就看不明白了

**Reply by shnehna** [reply]
哈哈哈 sorry 我就是把订阅地址导入进去后解析失败，
https://v2.xn--bnq37rc0nw50b.com/api/v1/client/subscribe?我的这个地址是不是不支持v2啊，之前用的都是clash系的，之前也一直发现v2系列的软件不支持，但是不知道这个专业术语怎么说。大佬你知道怎么解决吗

**Reply by LaoYutang** [reply]
你得看下机场有没有v2的配置订阅链接，配置的格式不一样的，一般机场都是支持很多种类型的，除非很远古不更新的机场了。

**Comment by hjkl950217** [comment]
GFW的规则好像不能更新了，不知道v2ray后面怎么解决。。。它默认的是访问“https://github.com/v2rayA/dist-v2ray-rules-dat/releases”下面的，现在一个也没了

**Reply by LaoYutang** [reply]
浏览器的话ZeroOmega吧

**Reply by hjkl950217** [reply]
> 浏览器的话ZeroOmega吧

主要是给非win的东西用，比如游戏主机这种不方便手动切换的。 浏览器上好说的，可惜了

**Reply by LaoYutang** [reply]
新版现在可以手动输入geo文件的地址了

**Reply by hjkl950217** [reply]
> 新版现在可以手动输入geo文件的地址了

soga

**Comment by shi-zaozui** [comment]
额，想问一下，这个v2raya，在哪儿配置代理规则啊？我是用clash for windows的那种，没看到v2raya在哪儿配置啊。
另外，clash for windows可以自动选择线路，这个可以这么配置吗

**Reply by LaoYutang** [reply]
1. 规则是指分流规则么，分流规则在设置里面，有个***规则端口的分流模式***，选择RoutingA就可以自定义规则。
2. 并没有自动线路选择，不过有多节点自动负载均衡，会找延迟最小的节点使用。

**Reply by shi-zaozui** [reply]
额，非常感谢，我这个部署上去是英文的，routingA找到在哪了，多节点自动负载均衡是在哪儿呢
![Uploading Snipaste_2025-02-06_16-32-56.png…]()

**Reply by LaoYutang** [reply]
你的图片打不开，右上角点用户名，里面的国旗可以改语言。
规则配置在设置的第二栏，可以参考[文档](https://v2raya.org/docs/manual/routinga/)

**Reply by LaoYutang** [reply]
对了负载均衡得是 2.2.4 版本的，需要的话要限制一下tag，直接连接多个节点就可以启用了。高版本的不知道为啥不行。

**Comment by Yakumo127** [comment]
局域网共享在哪里????

**Reply by Yakumo127** [reply]
是端口分享吗？？

**Reply by LaoYutang** [reply]
在设置的地址与端口选项里

**Reply by Yakumo127** [reply]
<img src="https://p.sda1.dev/21/68f42343874e319ca7412939366a348a/image.png" >
大佬是这个吗？开启这三个api吗？我开启后，还是不行

**Reply by LaoYutang** [reply]
不用开api服务，你只是要局域网使用这个代理端口吧。直接吧http代理地址配成``` ip:20171 ```就行

**Reply by LaoYutang** [reply]
注意：这个配置的代理端口需要docker run的时候映射出去，我上面映射的是7890端口，你现在用的20171端口

**Reply by Yakumo127** [reply]
是的，后面映射出去可以了，谢谢大佬。

**Reply by 1977499298** [reply]
> 不用开api服务，你只是要局域网使用这个代理端口吧。直接吧http代理地址配成`ip:20171`就行

这个ip是就用docker分配的ip吗，还是主机ip啊

**Reply by LaoYutang** [reply]
端口映射到主机了，局域网里面用主机的ip

**Reply by hjkl950217** [reply]
> > 不用开api服务，你只是要局域网使用这个代理端口吧。直接吧http代理地址配成`ip:20171`就行
> 
> 这个ip是就用docker分配的ip吗，还是主机ip啊

是主机ip.  比如  你是  容器:20171 ->  主机:7089     局域网访问的应该是  主机:7089

**Reply by Yakumo127** [reply]
不知道为什么 ，现在重新拉取后提示容器只监听 2017端口了，
root@bayun:~# docker logs v2raya | grep -i "listening"
2025/11/27 07:10:03.971 [A] [index.go:117]  v2rayA is listening at http://127.0.0.1:2017
2025/11/27 07:10:03.973 [A] [index.go:117]  v2rayA is listening at http://172.26.0.2:2017
2025/11/27 07:10:03.974 [A] [index.go:117]  v2rayA is listening at http://[::1]:2017
操作按照教程来的，7890端口和默认的20171 端口也放出来了

**Comment by Bruce-Sakura** [comment]
使用docker compose启动容器导入的链接无法获取全部节点，并且启动代理的时候会提示权限问题.

2025/03/08 01:57:59.050 [W] [transparent.go:55]  writeTransparentProxyRules: not support "redirect" mode of transparent proxy: ExecCommands: iptables-legacy -w 2 -t nat -N TP_OUT iptables v1.8.11 (nf_tables): Could not fetch rule set generation id: Permission denied (you must be root)
: exit status 4
2025/03/08 01:57:59.176 [E] [connection.go:88]  failed to start v2ray-core: not support "redirect" mode of transparent proxy: ExecCommands: iptables-legacy -w 2 -t nat -N TP_OUT iptables v1.8.11 (nf_tables): Could not fetch rule set generation id: Permission

网页上的提示
failed to start v2ray-core: not support "redirect" mode of transparent proxy: ExecCommands: iptables-legacy -w 2 -t nat -N TP_OUT iptables v1.8.11 (nf_tables): Could not fetch rule set generation id: Permission denied (you must be root) : exit status 4

**Reply by Flying-fat-cat** [reply]
俺也一样 failed to start v2ray-core: not support "redirect" mode of transparent proxy: ExecCommands: iptables-legacy -w 2 -t nat -N TP_OUT iptables v1.8.11 (nf_tables): Could not fetch rule set generation id: Permission denied (you must be root) : exit status 4

**Comment by majiabin375** [comment]
使用群晖的NAS部署docker成功了，但是每次添加订阅的时候总是报错，换了机场还是不行
failed to resolve subscription address: Get "http://link.dingyueapi.net/api/v1/client/subscribe?token=***": dial tcp: lookup link.dingyueapi.net: i/o timeout

**Reply by LaoYutang** [reply]
检查下dns的地址是否正常吧

**Comment by ShiroganeMiyuki** [comment]
大佬您好，我想问一下，如果是爱快docker部署，使用docker desktop拉取镜像并导出tar再导入爱快，在爱快新建容器设置里除了映射文件目录、端口还有啥要设置吗，我新建后没法访问，应该是没启动成功。导出的镜像tar只有4.8MB
